### PR TITLE
Workaround poll on macos.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
cc @rongou .

File: rabit/src/allreduce_base.cc [hunk 1]
```diff
@@ -549,7 +549,7 @@ AllreduceBase::TryAllreduceTree(void *sendrecvbuf_,
       break;
     }
     // select must return
-    auto poll_res = watcher.Poll(timeout_sec);
+    auto poll_res = watcher.Poll(timeout_sec, false);  // fail on macos
```
Comment (rongou) [inline rabit/src/allreduce_base.cc:552]: What does "fail on macox" mean? The old code was failing or the new code?

Comment (trivialfis) [reply rabit/src/allreduce_base.cc:552]: The old code is failing but did not check. The new code checks for those errors https://github.com/dmlc/xgboost/blob/8c676c889d380a32ed61ee2b69596c3f7ef154b8/rabit/include/rabit/internal/socket.h#L96 .

Comment (trivialfis) [reply rabit/src/allreduce_base.cc:552]: It seems those errors don't affect the actual output on macos. The underlying reason however, is unclear to me. Maybe I should try to get a mac device somehow.

Comment (rongou) [reply rabit/src/allreduce_base.cc:552]: You can start a macos instance on aws.

**Review by rongou — APPROVED** [review]

File: rabit/src/allreduce_base.cc (hunk 1 above)
Comment (trivialfis) [reply rabit/src/allreduce_base.cc:552]: I will make more checks for macos once the old rabit code can be replaced.

**Pull request merged**